dart/lib/src/flx_reader.dart

```diff
@@ -0,0 +1,416 @@
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:typed_data';
+import 'flx_types.dart';
+
+/// Main class to read a value out of a FlexBuffer.
+class FlxValue {
```

Comment:
Do we need a prefix like `Flex` here, because we don't have (or use) namespaces? If we can have namespaces, using names consistent with the other implementation, in particular the C++ one, would be good (this one would be `Reference` ?)

Reply:
No its not necessary, it is just something that I used. I can rename the type to `Reference` and dart also has the `as` keyword in combination with `import` so users can rename things as they are pleased, or in case there are name collisions.